library/kotlin/io/envoyproxy/envoymobile/EngineBuilder.kt

```diff
@@ -162,7 +163,7 @@ open class EngineBuilder(
    *
    * @return this builder.
    */
-  fun addStringAccessor(name: String, accessor: () -> String): EngineBuilder {
+  fun addStringAccessor(name: String, accessor: () -> ByteBuffer): EngineBuilder {
```

Comment:
IMO the public interface should still return a String. We can map it to ByteBuffer behind the scenes.

Reply:
Also why not use the ByteArray directly? We can't predict the lifetime of its usage, so we have to do a copy anyways - might as well just do it via the critical array copy that we already have.

Reply:
@goaway I feel like we can revisit our ByteBuffer versus ByteArray conversation again. Though I do agree if we can have a String interface but internally, it just transforms to a ByteBuffer, that would be nice

Reply:
My point though is that we're converting to a byte array and then wrapping with a byte buffer when we already support passing byte array directly.

Reply:
@buildbreaker @goaway updated. Pretty pleased with how it came out. Easy and ergonomic for the user, and consistent with prior JNI experience under the hood. Lmk what you think.